### PR TITLE
Patient/medicinenotification

### LINKED ISSUE #45: Patient: Medicine consumation notification

### PR DESCRIPTION
**Pull request by VladimirGolosin**
Changes:
-Added functionality to see notifications of when to take prescriptions
closes #45

### CODE REVIEW

File: HospitalSystem.ConsoleUI/PatientUI.cs [hunk 1]
```diff
@@ -983,8 +988,75 @@ public void StartDoctorSearch()
...
+                    break;
+                    case MedicationBestTaken.BEFORE_MEAL:
+                    meal = "before meal";
+                    break;
+                    case MedicationBestTaken.ANY_TIME:
+                    meal = "any time";
+                    break;
+                    case MedicationBestTaken.WITH_MEAL:
+                    meal = "with meal";
+                    break;
```
Comment (swopl) [inline HospitalSystem.ConsoleUI/PatientUI.cs:1034]: Bad indents for case statement

**Review by KatarinaM-SV6 — APPROVED** [review]
Seems okay

File: HospitalSystem.Core/People/Patient/PatientService.cs [hunk 2]
```diff
@@ -94,4 +94,33 @@ public Patient GetPatientById(ObjectId id)
...
+        DateTime beginsToTake = new DateTime(now.Year, now.Month, now.Day, 8, 0, 0);
+        List<DateTime> times = new List<DateTime>();
+        for (int i=0; i<prescription.TimesADay; i++)
+        {
+            DateTime timeToTake = beginsToTake.AddHours(i*prescription.HoursBetweenIntakes);
+            times.Add(timeToTake);
+        }
+
+        return times;
+    }
```
Comment (swopl) [inline HospitalSystem.Core/People/Patient/PatientService.cs:111]: Does not belong in this service. Also could use a better name

File: HospitalSystem.Core/People/Patient/PatientService.cs [hunk 3]
```diff
@@ -94,4 +94,33 @@ public Patient GetPatientById(ObjectId id)
...
+        List<DateTime> timesForMedicine = GetAllTimesForMedicine(prescription);
+        foreach (DateTime time in timesForMedicine)
+        {
+            if (DateTime.Now < time && DateTime.Now.Add(patient.WhenToRemind)>= time)
+            {
+                return time;
+            }
+        }
+        return null;
+    }
```
Comment (swopl) [inline HospitalSystem.Core/People/Patient/PatientService.cs:125]: Prob should not be in this service

File: HospitalSystem.Core/People/Patient/Patient.cs [hunk 4]
```diff
@@ -6,12 +6,21 @@ public class Patient : Person
...
             LastName = lastName;
             MedicalRecord = medicalRecord;
             CheckupChangeLogs = new List<CheckupChangeLog>();
+
+            if (whenToRemind is null){
+                WhenToRemind = TimeSpan.FromHours(1);
+            }
+            else{
+                WhenToRemind = (TimeSpan)whenToRemind;
+            }
```
Comment (swopl) [inline HospitalSystem.Core/People/Patient/Patient.cs:22]: Why do you allow null passing? When does that happen?

File: HospitalSystem.ConsoleUI/PatientUI.cs [hunk 5]
```diff
@@ -983,8 +988,75 @@ public void StartDoctorSearch()
...
+        int numberOfMinutes;
+        Console.WriteLine("Please enter how many minutes before perscription should be considered (min 5, max 300): ");
+        try
+        {
+            numberOfMinutes = ReadInt(5, 300, "Number out of bounds!", "Number not recognized!");
+        }
+        catch (InvalidInputException e)
+        {
+            System.Console.Write(e.Message + " Aborting...");
+            throw new QuitToMainMenuException("Wrong input");
```
Comment (swopl) [inline HospitalSystem.ConsoleUI/PatientUI.cs:1055]: Wrong exception type. QTMM should be used for user driven quitting

**Review by swopl — CHANGES_REQUESTED** [review]

File: HospitalSystem.ConsoleUI/PatientUI.cs (hunk 1 above)
Comment (VladimirGolosin) [reply HospitalSystem.ConsoleUI/PatientUI.cs:1034]: will fix

File: HospitalSystem.Core/People/Patient/PatientService.cs (hunk 2 above)
Comment (VladimirGolosin) [reply HospitalSystem.Core/People/Patient/PatientService.cs:111]: Any ideas?

Comment (VladimirGolosin) [reply HospitalSystem.Core/People/Patient/PatientService.cs:111]: Should i make medicine notification service and place it there?

File: HospitalSystem.Core/People/Patient/Patient.cs (hunk 4 above)
Comment (VladimirGolosin) [reply HospitalSystem.Core/People/Patient/Patient.cs:22]: This is just a way to make default value to WhenToRemind, i would do it in parameters but it creates an error

File: HospitalSystem.ConsoleUI/PatientUI.cs (hunk 5 above)
Comment (VladimirGolosin) [reply HospitalSystem.ConsoleUI/PatientUI.cs:1055]: I'm aware of that, i'll fix it later.

File: HospitalSystem.Core/People/Patient/PatientService.cs (hunk 2 above)
Comment (VladimirGolosin) [reply HospitalSystem.Core/People/Patient/PatientService.cs:111]: VSCode is acting weird, i won't be able to test it...

**Review by swopl — APPROVED** [review]

File: HospitalSystem.Core/People/Patient/PatientService.cs (hunk 2 above)
Comment (swopl) [reply HospitalSystem.Core/People/Patient/PatientService.cs:111]: This is smart code. Maybe put in Prescription? Idk what that class looks like....

**Pull request merged**